crates/iota-graphql-client/src/lib.rs

```diff
@@ -1477,12 +1478,44 @@ impl Client {
             let bcs = base64ct::Base64::decode_vec(result.effects.bcs.0.as_str())?;
             let effects: TransactionEffects = bcs::from_bytes(&bcs)?;
 
+            if wait_for_finalization {
+                self.wait_for_tx_finalization(tx.digest(), None).await?;
+            }
+
             Ok(Some(effects))
         } else {
             Ok(None)
         }
     }
 
+    /// Wait for the finalization of a transaction by its digest. An optional
+    /// timeout can be provided, which, if exceeded, will return an error.
+    /// Returns the [`TransactionEffects`].
+    pub async fn wait_for_tx_finalization(
+        &self,
+        digest: Digest,
+        timeout: impl Into<Option<Duration>>,
+    ) -> Result<Option<TransactionEffects>> {
```

Comment:
No need to return an Option if we return an error because of the timeout otherwise?